src/libDirectoryService/DirectoryService.h

```diff
@@ -86,7 +86,11 @@ class DirectoryService : public Executable, public Broadcastable
     std::shared_ptr<DSBlock> m_pendingDSBlock;
     std::mutex m_mutexPendingDSBlock;
     std::mutex m_mutexDSBlockConsensus;
-    std::vector<std::pair<PubKey, boost::multiprecision::uint256_t>> m_allPoWs;
+    //nonce + winning hash pair
+    std::vector<std::pair<
+        PubKey,
+        pair<boost::multiprecision::uint256_t, std::array<unsigned char, 32>>>>
+        m_allPoWs;
```

Comment:
Multiple pair can use tuple instead, consider the below definition?
using PoW = std::tuple<PubKey, boost::multiprecision::uint256_t, std::array<unsigned char, 32>>;
using VectorOfPoW = std::vector<PoW>;
VectorOfPoW m_allPoWs;